core/src/main/java/com/datastax/oss/driver/api/core/metadata/Node.java

```diff
@@ -178,10 +178,10 @@
    * The host ID that is assigned to this node by Cassandra. This value can be used to uniquely
    * identify a node even when the underling IP address changes.
    *
-   * <p>This should be non-null in a healthy deployment, but the driver will still function, and
-   * report {@code null} here, if the server metadata was corrupted.
+   * <p>This information is required; the driver will not function properly if this method returns
+   * {@code null}.
```

Comment:
The user doesn't provide `Node` instances, the driver does, so I think we should rather formulate this as "this information is always present" or something similar.